### ISSUE TITLE
Why is the torch computation graph kept when training with rollout?

### ISSUE THREAD

**Issue by radiradev** [ecmwf/anemoi-core#419]
When training with rollout the autograd graph is saved for each indivual rollout step.
 https://github.com/ecmwf/anemoi-core/blob/665f4629649d2852449059ea2cc7af9bad7e24da/training/src/anemoi/training/train/forecaster/forecaster.py#L395-L410 
Couldn't we instead call `.backward()` at every rollout step and call step the optimizer at the end of rollout for the same training dynamics but we reduced memory (and compute due to checkpointing)?

So in pseudo-code:

```
for step in rollout_steps:
    loss = compute_loss(self(x) - truth)
    loss.backward() #saves gradient

optimizer.step()
```

It's a bit more complicated to do since `anemoi` uses `pytorch-lightning` but there are options for this here https://lightning.ai/docs/pytorch/stable/api/lightning.pytorch.core.LightningModule.html#lightning.pytorch.core.LightningModule.manual_backward

**Comment by japols** [comment]
Good idea! If you're up for it, feel free to open a PR — happy to help review/benchmark.

**Comment by ssmmnn11** [comment]
Would be nice to have the manual backward option / example.

My recommendation would be: implement the current training logic with manual backward, then check that everything is identical (we also want to keep the option of gradient clipping and accumulation, mixed precision etc.). If all looks good / identical try the next step. :-)

**Comment by ssmmnn11** [comment]
@radiradev are you working on this? If not, I could come up with a first draft. :-)
